### ISSUE TITLE
Please add option to normalize the range of greys across the whole cut depth and another to ignore black

### ISSUE THREAD

**Issue by charlieb** [jes/pngcam#1]
In gimp I can adjust the brightness so that the histogram is all the way towards the black spike on the left hand side, then adjust the levels to smear the histogram back out across the whole range of greys. If I don't do this the image can either be:
too dark - carve will have some unnecessary blank space at the top where there will just be a clear cut down to a certain depth before any detail is cut or
too light - all the detail in the image will be at the very top with deep cliffs around the cut where the images becomes black.

Unfortunately I don't speak perl but some preprocessing pseudocode might look like:
```

high = 0
low = 255
for every pixel:
  if pixel is black and ignore black:
    continue
  if pixel > high: high = pixel
  if pixel < low: low = pixel

function mapping(pixel): 

for every pixel:
    if pixel is black and ignore black:
        continue
    set pixel = (pixel-low) * 255 / high-low
```

Thanks

**Comment by jes** [comment]
Good idea. I can see the benefit of normalising the colours. I don't quite see what you mean by "ignore black"?
Do you mean you don't want any of the black to be cut at all? Or you want black to remain black even while the rest of the colours are normalised?

Do you have any example heightmaps that would help illustrate what you're trying to do?

**Comment by charlieb** [comment]
I think just ignore black for the purposes of calculating the range for
normalization. Once the rest of the pixels have been normalized black will
just naturally be the lowest point of the cut as normal. That logic is
captured in the pseudocode. Once the preprocessing is done the two option
variables will no longer be needed and the rest of your excellent program
can proceed unchanged.

**Comment by jes** [comment]
@charlieb I've implemented this in bf62e0b, hopefully that does what you want. The options are `--normalise` and `--normalise-ignore-black`, for want of a better name.
